docs/web/authentication.md

```diff
@@ -320,6 +323,117 @@ groups. For more information [see](permissions.md#managing-permissions).
 
 ----
 
+### <i>OAuth</i> authentication <a name="oauth-authentication"></a>
+
+CodeChecker also supports OAuth-based authentication. The `authentication.method_oauth` section contains the configuration for OAuth authentication for different OAuth providers. The server can be configured for different Oauth `providers`. Users can be added into the `allowed_users`.
+
+#### OAuth Configuration options <a name="oauth-configuration-options"></a>
+  * `enabled`
+
+    Indicates if OAuth authentication is enabled (required for any methods below)
+
+ * `providers`
+
+    The provider field contains configuration details for OAuth providers. Each provider's configuration includes but may vary depending on provider:
+
+  * `provider_name` as an object containing following properties:
+
+      * `enabled`
+
+          Indicates if current provider is enabled (github, google, etc.)
+
+      * `client_id`
+
+           Contains client ID provided by the OAuth provider.
+
+      * `client_secret`
+
+          The client secret must be provided by the OAuth provider.
+
+      * `authorization_url`
+
+          This link is used for redirecting user for provider's authentication page
+
+      * `callback_url`
+
+          User will be redirected back to the provided link after login with returned data.
+          It should be constructed in that format `http://codechecker_path/login/OAuthLogin/provider` where `provider` is the name of the provider of OAuth and should match existing `provider_name`. The `callback_url` should also match the callback url specified in the provider's configuration.
+
+          Example of correct link using github, google and microsoft
+          * http://localhost:8080/login/OAuthLogin/github
+          * http://localhost:8080/login/OAuthLogin/google
+          * http://localhost:8080/login/OAuthLogin/microsoft
+
+      * `token_url`
+
+          The URL to exchange the authorization code for an access token.
+
+      * `user_info_url`
+
+          The URL to fetch the authenticated user's information.
+
+      * `user_emails_endpoint`
+
+          The URL is used for making requests for emails associated with github account.
+          This field only in relevant for github in current implementation.
+
+      * `scope`
+
+          The scope of access requested from the OAuth provider.
+
+      * `user_info_mapping`
```

Comment:
I have a feeling that this adds unnecessary extra complexity to the logic. Why do we want to use the e-mail address as user name? Why couldn't the auth provider decide what identifies the user? This way it wouldn't be needed to query the e-mail address at different API endpoints in GitHub. I can't see why e-mail address is important from authentication point of view. Let's discuss these questions in a separate meeting.